lib/wasi/src/runtime/resolver/builtin.rs

```diff
@@ -0,0 +1,66 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use url::Url;
+
+use crate::{
+    bin_factory::BinaryPackage,
+    http::HttpClient,
+    runtime::resolver::{types::ResolverError, types::WebcIdentifier, PackageResolver},
+};
+
+/// The builtin package resolver, backed by WAPM.
+///
+/// Any downloaded assets will be cached on disk.
+#[derive(Debug, Clone)]
+pub struct BuiltinResolver {
```

Comment:
`Builtin` is a suboptimal name, it doesn't really express what it does. 

Maybe `RegistryResolver` ?

Reply:
Renamed.